File: angularAuthorizationExample/Models/MaintenanceModel.cs
namespace angularAuthorizationExample.Models
{
    public class MaintenanceModel
    {
        public int IdVehicle {get; set;}
        public int Id {get; set;}
        //ToDo: Change it to string type
        public string MaintenanceDate{get; set;}    
        public Decimal Mileage{get; set;}
        public string? Description{get; set;}
        public decimal Cost{get; set;}
        public int Distance { get; set; }
        public List<MaintenanceDetailsModel>?MaintenanceDetailsList{get; set;}
    }
}

File: angularAuthorizationExample/ClientApp/src/app/maintenances/MaintenanceModel.ts
import { IMaintenanceDetailsAngularModel } from "./MaintenanceDetailsAngularModel";

export interface IMaintenanceAngularModel {
    id: number;
    idVehicle: number;
    maintenanceDate: string;
    mileage: number;
    distance: number;
    cost: number;
    description: string;
    errorMessage: string;
    maintenanceDetailsList: IMaintenanceDetailsAngularModel[];
  }
  
